### PR TITLE
Provide user defined credentials for token based auth methods

### PR DESCRIPTION
**Pull request by pdziok**

### CODE REVIEW

**Review by charlie47 — COMMENTED** [review]
How about extending the readme file with some info?

**Review by charlie47 — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -69,9 +71,11 @@ const registerBearerStrategy = function(startWith, server, options) {
             server.auth.strategy('bearer', 'bearer-access-token', {
                 validateFunc: (token, callback) => {
                     if (options.tokens.hasOwnProperty(token)) {
-                        return callback(null, true, {
-                            newsroom: options.tokens[token]
-                        });
+                        let credentials = {newsroom: options.tokens[token]};
+                        if (typeof options.credentials === 'object') {
```
Comment (kubek2k) [inline index.js]: I am not sure if that check has to be here - 🐻  in mind that if `credentials` is not an object the code will silently not do anything (which is probably not what we want). So either a proper validation (supposedly somewhere on top of `registerBearerStrategy` function or no validation (and letting it crash).

Comment (kubek2k) [reply index.js]: and - `additionalCredentials` name would be better (since those are mixed in).

**Review by kubek2k — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (pdziok) [reply index.js]: Data validation is handled by Joi. Name changed

**Comment by kubek2k** [comment]
I think proposal to extend README.md is quite valid

**Comment by pdziok** [comment]
I totally missed it. updating readme then 👍

**Review by kubek2k — APPROVED** [review]

**Pull request merged**